Terminal.Gui/Windows/FileDialog2.cs

```diff
@@ -0,0 +1,2118 @@
+		public IReadOnlyList<string> MultiSelected { get; private set; }
+
+
+		/// <inheritdoc/>
+		public override void Redraw (Rect bounds)
+		{
+			base.Redraw (bounds);
+
+			this.Move (1, 0, false);
+
+
+			if (ustring.IsNullOrEmpty (Title)) {
+				return;
+			}
+
+			var title = this.Title.ToString ();
+			var titleWidth = title.Sum (c => Rune.ColumnWidth (c));
+
+			if (titleWidth > bounds.Width) {
+				title = title.Substring (0, bounds.Width);
+			} else {
+				if (titleWidth + 2 < bounds.Width) {
+					title = '╡' + this.Title.ToString () + '╞';
+				}
+				titleWidth += 2;
+			}
+
+			var padLeft = ((bounds.Width - titleWidth) / 2) - 1;
+
+			padLeft = Math.Min (bounds.Width, padLeft);
+			padLeft = Math.Max (0, padLeft);
+
+			var padRight = bounds.Width - (padLeft + titleWidth + 2);
+			padRight = Math.Min (bounds.Width, padRight);
+			padRight = Math.Max (0, padRight);
+
+			Driver.SetAttribute (
+			    new Attribute (this.ColorScheme.Normal.Foreground, this.ColorScheme.Normal.Background));
+
+			Driver.AddStr (ustring.Make (Enumerable.Repeat (Driver.HDLine, padLeft)));
+
+			Driver.SetAttribute (
+			    new Attribute (this.ColorScheme.Normal.Foreground, this.ColorScheme.Normal.Background));
+			Driver.AddStr (title);
+
+			Driver.SetAttribute (
+			    new Attribute (this.ColorScheme.Normal.Foreground, this.ColorScheme.Normal.Background));
+
+			Driver.AddStr (ustring.Make (Enumerable.Repeat (Driver.HDLine, padRight)));
+		}
+
+		/// <inheritdoc/>
+		public override void OnLoaded ()
+		{
+			base.OnLoaded ();
+
+			// if filtering on file type is configured then create the ComboBox and establish
+			// initial filtering by extension(s)
+			if (this.AllowedTypes.Any ()) {
+
+				this.currentFilter = this.AllowedTypes [0];
+				var allowed = this.AllowedTypes.ToList ();
+
+				if (!this.AllowedTypesIsStrict) {
+					allowed.Insert (0, AllowedType.Any);
+				}
+
+				// +2 to allow space for dropdown arrow
+				var width = this.AllowedTypes.Max (a => a.ToString ().Length) + 2;
+
+				var combo = new ComboBox (allowed) {
+					Width = width,
+					ReadOnly = true,
+					Y = 1,
+					X = Pos.AnchorEnd (width),
+					Height = allowed.Count + 1,
+					SelectedItem = this.AllowedTypesIsStrict ? 0 : 1
+				};
+
+				combo.SelectedItemChanged += (e) => this.Combo_SelectedItemChanged (combo, e);
+
+				this.Add (combo);
+				this.LayoutSubviews ();
+			}
+
+			// if no path has been provided
+			if (this.tbPath.Text.Length <= 0) {
+				this.tbPath.Text = Environment.CurrentDirectory;
+			}
+
+			// to streamline user experience and allow direct typing of paths
+			// with zero navigation we start with focus in the text box and any
+			// default/current path fully selected and ready to be overwritten
+			this.tbPath.FocusFirst ();
+			this.tbPath.SelectAll ();
+
+			if (ustring.IsNullOrEmpty (Title)) {
+				switch (OpenMode) {
+				case OpenMode.File:
+					this.Title = $"OPEN {(MustExist ? "EXISTING " : "")}FILE";
```

Comment:
Localalizable?

Let's debate `ALLCAPS` vs not for dialog titles. If we agree we like `ALLCAPS` stylistically, let's build `ToUpper` into the new `View.Title`.  And make it configurable!